src/order/hom/basic.lean

```diff
@@ -683,3 +683,22 @@ theorem order_iso.is_complemented_iff :
 
 end bounded_order
 end lattice_isos
+
+section boolean_algebra
+variables (α) [boolean_algebra.core α]
+
+/-- Taking complements as an order isomorphism to the order dual. -/
+def order_iso.compl : α ≃o order_dual α :=
```

Comment:
You're missing the simp lemmas. Either you tag ` order_iso.compl` with `@[simps]` and you fix `to_fun` and `inv_fun` so that they have the correct type (instead of `α → α`) or you write the simp lemmas yourself.